### PR TITLE
fix: manifest name does not need to match url

### LINKED ISSUE #33: Manifest Name Restriction
It never made sense to me why the manifest names have to match the repository name to verify that they are installed. Just take the URL from the repository name instead.

### PR DESCRIPTION
**Pull request by 0x4007**
Resolves #33

QA:

Before, check assistive pricing not being "installed" [in @ubiquity org](https://github.com/ubiquity/.ubiquity-os/blob/development/.github/.ubiquity-os.config.yml). 

IMAGE 2024-12-10 22:35:48

After it is now "installed"
image

### CODE REVIEW

**Comment by 0x4007** [comment]
- Testing pushing `wallet` changes: https://github.com/ubiquity/.ubiquity-os/commit/35920b13cf364b9ca74b37e2b157a4e2e1e47133
- And according to the help menu two instances of wallet are installed: https://github.com/ubiquity/.github-private/issues/2#issuecomment-2532695486

`development` and `main` are now installed. This seems out of scope to fix here. 

```yml
- plugin: https://ubiquity-os-command-wallet-main.ubiquity.workers.dev
- plugin: https://ubiquity-os-command-wallet-development.ubiquity.workers.dev
```

I just tested against the latest stable deploy at https://22e3a662.ubiquity-os-plugin-installer.pages.dev/ and the problem persists so I will not investigate as part of this pull. I am not sure why it retains the development instance of the plugin.

Test result: https://github.com/ubiquity/.ubiquity-os/commit/135f15b7d8c321c73d9d98dac2457c0f46008029

**Comment by 0x4007** [comment]
After reverting the config, I redid the test and the `development` version of wallet continues to be selected. I am now assuming that it takes the default branch of each plugin, and makes it not recognize the `-main` version of wallet in the config. https://github.com/ubiquity/.ubiquity-os/commit/4350111e8d623c1e7f1260d81aae6d9fa8d75313

Interesting enough, it still says its installed after using the UI remove function https://github.com/ubiquity/.ubiquity-os/commit/8c4eee0495071c79ba983bb27c1f96117f875fbe

image


Perhaps it makes sense to change all of our default branches in the @ubiquity-os-marketplace organization to `main`

**Review by rndquu — APPROVED** [review]

File: static/scripts/rendering/config-editor.ts [hunk 1]
```diff
@@ -84,8 +85,36 @@ export function renderConfigEditor(renderer: ManifestRenderer, pluginManifest: M
   }
 
   const parsedConfig = renderer.configParser.parseConfig(renderer.configParser.repoConfig || localStorage.getItem("config"));
-  // for when `resetToDefault` is called and no plugin gets passed in, we still want to show the remove button
-  const isInstalled = parsedConfig.plugins?.find((p) => p.uses[0].plugin.includes(normalizePluginName(pluginManifest?.name || "")));
```
Comment (0x4007) [inline static/scripts/rendering/config-editor.ts:88]: This is unclear to me what this is solving @Keyrxng

**Comment by rndquu** [comment]
> Perhaps it makes sense to change all of our default branches in the https://github.com/ubiquity-os-marketplace organization to `main`

I suppose when https://github.com/ubiquity-os/plugin-template/issues/33 is ready it would be way easier to check what plugins are installed

File: static/utils/strings.ts [hunk 2]
```diff
@@ -6,3 +6,21 @@ export const STRINGS = {
...
+export function extractPluginIdentifier(url: string): string {
+  // For GitHub manifest URLs, extract just the repo name
+  if (url.includes("github.com/") || url.includes("githubusercontent.com/")) {
+    const parts = url.split("/");
+    if (parts.length >= 5) {
+      return parts[4].split("@")[0].split("?")[0]; // Get repo name without branch or query params
+    }
+  }
+
+  // For all other URLs (workers, actions, etc), use the full URL
```
Comment (0x4007) [inline static/utils/strings.ts]:
```suggestion
  // For all other URLs (workers, etc), use the full URL
```

File: static/utils/strings.ts [hunk 3]
```diff
@@ -6,3 +6,21 @@ export const STRINGS = {
   SELECT_ARROW_ACTIVE: "select-arrow-active",
   PICKER_SELECT: "picker-select",
 };
+
+/**
+ * For manifest URLs from GitHub, extracts just the repo name.
+ * For all other URLs (workers, actions, etc), returns the full URL.
```
Comment (0x4007) [inline static/utils/strings.ts]:
```suggestion
 * For all other URLs (workers, etc), returns the full URL.
```

File: static/scripts/rendering/config-editor.ts (hunk 1 above)
Comment (Keyrxng) [reply static/scripts/rendering/config-editor.ts:88]: The comment is pretty clear I thought, idk how else to explain it but this was Zug's review comment on it:

> pressing reset to default is disabling the remove button when it shouldn't please mind that you should pass in plugin to renderConfigEditor

**Pull request merged**